oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+   this by providing a responsive grid system along with components that
+   work equally well on mobile and web.
+
+4. **Designs should be fully accessible**
+
+   Rationale: Open edX measures and evaluates accessibility using the World
+   Wide Web Consortium’s `Web Content Accessibility Guidelines (WCAG) 2.0`_.
+   All features merged into edX repositories are expected to conform to
+   Level AA of this specification and to satisfy the requirements outlined in
+   the `edX Website Accessibility Policy`_.
+
+   You can learn more about developing accessible applications with
+   the Developer's Guide chapter `edX Accessibility Guidelines`_.
+
+5. **Front end applications should support localization**
+
+   Rationale: Open edX is used by learners all over the world, and so
+   it is critical that all new functionality support localization. This
+   includes ensuring that all new user interfaces will switch to
+   right-to-left mode when the language requires it.
+
+   You can learn more about localization with the Developer's Guide
+   chapter `edX Internationalization`_.
+
+7. **Front end implementations should support theming**
+
+   Rationale: Open edX is used by organizations around the world,
+   and each one requires that the user experience is based upon their
+   own brand. This includes elements as simple as overriding the fonts,
+   imagery and color schemes, to more complex customization of headers,
+   footers and registration flows.
+
+   You can learn more about how Open edX supports theming with
+   the guide `Changing Themes for an Open edX Site`_. One simple way to
+   make your interface themeable is to use patterns from the
+   `edX Pattern Library`_ as they already support theming.
+
+8. **Front end code should be safe from cross site scripting**
+
+   Cross Site Scripting (XSS) vulnerabilities allow user-supplied data to be
+   incorrectly executed as code in a web browser. This can allow an attacker
+   to inject malicious scripts, which in the worst case can cause the loss
+   of private user data.
+
+   In order to keep your code safe from such attackers, you should follow the
+   best practices laid out in the guide
+   `Preventing Cross Site Scripting Vulnerabilities`_.
+
+Technology Selection
+~~~~~~~~~~~~~~~~~~~~
+
+1. **Server-side templating should use Django Templates**
+
+   Rationale: There are many template languages available for Django,
+   but the simplest option is to use the built-in Django template engine.
+   The Open edX codebase has a mixture of Django and Mako templates, but the
+   former are easier to reason about because they don't support arbitrary
+   code evaluation. See `Reasons for rejecting Mako`_ for a more detailed
+   explanation.
+
+   **Exception**: Mako templates should be used within edx-platform for
+   consistency with the existing code base. Unfortunately there has been too
+   much investment in Mako to consider replacing it at this point.
```

Comment:
Django 1.8+ supports concurrent use of multiple template engines.  Is it worth investigation of a gradual transition to Django templates over time?